pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -666,14 +684,28 @@ func (p *PriorityQueue) isPodBackingoff(podInfo *framework.QueuedPodInfo) bool {
 
 // SchedulingCycle returns current scheduling cycle.
 func (p *PriorityQueue) SchedulingCycle() int64 {
-	p.lock.RLock()
-	defer p.lock.RUnlock()
+	p.activeQLock.RLock()
+	defer p.activeQLock.RUnlock()
 	return p.schedulingCycle
 }
 
+func (p *PriorityQueue) dumpClusterEvents(inFlightPod *list.Element) []*clusterEvent {
+	var events []*clusterEvent
+	for event := inFlightPod.Next(); event != nil; event = event.Next() {
+		e, ok := event.Value.(*clusterEvent)
+		if !ok {
+			// Must be another in-flight Pod (*v1.Pod). Can be ignored.
+			continue
+		}
+		events = append(events, e)
+	}
+	return events
+}
+
 // determineSchedulingHintForInFlightPod looks at the unschedulable plugins of the given Pod
 // and determines the scheduling hint for this Pod while checking the events that happened during in-flight.
 func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) queueingStrategy {
+	p.activeQLock.RLock()
```

Comment:
`RUnlock` everywhere like the current change could lead a future mistake forgetting unlock. Maybe it's much simpler to do something like 

```go
p.activeQLock.RLock()
unlocked := false
defer func() {
  if !unlocked {
    p.activeQLockRUnLock()
  }
}()

//at L740
p.activeQLockRUnLock()
unlocked = true
```

Reply:
Moved the code that requires lock to the separate function. I think it's clear now